src/dune_rules/lib.ml

```diff
@@ -411,7 +423,9 @@ and resolve_result =
   | Invalid of User_message.t
   | Ignore
   | Redirect_in_the_same_db of (Loc.t * Lib_name.t)
+  | Multiple_results of resolve_result list
   | Redirect of db * (Loc.t * Lib_name.t)
+  | Deprecated_library_name of (Loc.t * Lib_name.t)
```

Comment:
Adding a new variant to distinguish between a regular redirect (public libs) and deprecated libs. Treating them both the same way leads to all the tests in `test/blackbox-tests/test-cases/deprecated-library-name/features.t` failing because of duplicated errors. See related PR #10231 (those changes were added directly into this PR).